doc/template.txt

```diff
@@ -285,5 +285,16 @@ A: Please file an issue at https://github.com/aperezdc/vim-template/issues
    common questions may end up in the |template-troubleshooting| section of
    the documentation (this one you are reading).
 
+---
+
+Q: Why is only variable replaced per line?
+
+A: This plugin uses the vim's substitute with the "g" option, which means
+   all occurrences shall be removed.
+   If not, ensure that gdefault and edcompatible is off as
+   documented at http://vim.wikia.com/wiki/Substitute
+   The option gdefault causes the inversion of the "g" option which is
+   then in conflict with the intended behaviour of this plugin.
+
 
```

Comment:
Wow, I did not know before today about the `gdefault` setting. Indeed it's a good thing to mention this in the documentation.

Reply:
Me neither .... I started to fill an issue and was thinking what the solution could be.
After having a look at your code I found out that my configuration was wrong .... so I was further digging deeper into the configuration.